administrator/components/com_associations/models/fields/itemlanguage.php

```diff
@@ -72,17 +74,40 @@ protected function getOptions()
 				{
 					parse_str($associations[$lang->value], $contents);
 					$removeExtra  = explode(":", $contents['id']);
-					$lang->value  = $lang->value . "|" . $removeExtra[0];
+					$itemId       = $removeExtra[0];
+					$lang->value  = $lang->value . "|" . $itemId;
 				}
 				else
 				{
+					$itemId      = $associations[$lang->value];
 					$lang->value = $lang->value . "|" . $associations[$lang->value];
 				}
+
+				$canEdit    = $user->authorise('core.edit', $component->assetKey . '.' . $itemId);
+				$table->load($itemId);
+
+				if (!is_null($table->{$component->fields->created_by}))
+				{
+					$canEditOwn = $user->authorise('core.edit.own', $component->assetKey . '.' . $itemId) && $table->{$component->fields->created_by} == $user->id;
+					$canEdit    = $canEdit || $canEditOwn;
+				}
```

Comment:
IMHO this could be improved.
you only need to load the item with jtable (extra bd query) if the component supports created_by field, so  ...

```php
if (!is_null($component->fields->created_by))
{
	$table->load($itemId);
	$canEditOwn = $user->authorise('core.edit.own', $component->assetKey . '.' . $itemId) && $table->{$component->fields->created_by} == $user->id;
	$canEdit    = $canEdit || $canEditOwn;
}
```